### ISSUE TITLE
Callback back for angular template compiler

### ISSUE THREAD

**Issue by kumaresan-subramani** [angular/angular#41115]
# 🚀 Feature request

callback event for template rendering

### Command (mark with an `x`)

- [ ] new
- [ ] build
- [x] serve
- [ ] test
- [ ] e2e

### Description
I want to change my angular component dynamically, while doing this i am facing below issue

some times `<ng-template>` rendering gets delayed so i cant chnage that element

is there anyway to check whether `'ng-template` gets rendered or not

### Describe alternatives you've considered
Not yet

### sample

https://stackblitz.com/edit/angular-ivy-1igf7y?file=src%2Fapp%2Fapp.component.ts

How to find `ng-templete` rendered or not

**Comment by petebacondarwin** [comment]
It is not clear to me exactly what the use case is here. In your sample what change might you want to make and at what point?
Be aware that there is no template compilation going on here. The runtime is simply rendering the template, which will have been compiled at build time (or in the case of JIT compilation, during compilation of the containing component).

**Comment by kumaresan-subramani** [comment]
@petebacondarwin , Any suggestion or update on this.

**Comment by AleksanderBodurri** [comment]
@kumaresan-subramani This may have more to do with how ej2 handles converting html to PDF (**maybe it doesn't actually convert from html but instead just uses a default template with your data**?) than anything in the framework. You might have to investigate how the `pdfExport` method and the `PdfExportModule` from ej2 work to solve this issue.

**Comment by petebacondarwin** [comment]
Yeah - this scenario seems super complicated and it is impossible to debug into the example since the syncfusion library is minified. It is not clear to me why the PDF is not containing the expected output.

In any case, I don't believe that we will add a rendered event to the core, since that would negatively affect performance of a hot area of the runtime.

**Comment by JoostK** [comment]
Knowing when an ng-template has been rendered seems like a niche problem that can be achieved without framework support, e.g. using a custom directive similar to `NgTemplateOutlet` that emits events or calls into a service to notify that some template ref has been rendered. In the described situation it's unclear to me whether rendering of ng-templates is even the culprit of the problem you're experiencing.

In summary, I'm closing this request as not something we'll add to Angular.
